### PR TITLE
Devops/single page recipe logic (PRA-116)

### PR DESCRIPTION
**Pull request by AnnaPestova1**
## One Line Description
Add logic for single saved recipe page (filter by tags, delete, print recipe, all buttons active)

## Requirements
now we can filtered by tags on single recipe page (redirect on recipes page), added tags on recipes page and filtering through them. Added active filtering button. added delete option and print option on single recipe page, added page stubs to make all button functional

_What this code should do_
you should go to the recipes page, click on recipe card and, after redirecting on the single recipe page, try to push all buttons.

## Notes
fixed merge conflict with current devops/initial-setup branch (Aug, 11)
_Any additional info_

## Test Steps

_Instructions on how to test the changes_

## UI/UX Outcome

_Attach a screenshot here_

## Checklist

- The Linear ID is in the PR title
- My code follows best practices
- Documentation is included where needed

### CODE REVIEW

**Review by AnnaSolovykh — APPROVED** [review]
LGTM!

File: src/components/SavedRecipes.tsx [hunk 1]
```diff
@@ -10,73 +20,123 @@ const SavedRecipes = () => {
...
   const [filteredRecipes, setFilteredRecipes] = useState([]);
   const [searchParams, setSearchParams] = useSearchParams();
+  const [activeCategory, setActiveCategory] = useState("");
+  const [activeTag, setActiveTag] = useState("");
   const [show, setShow] = useState(true);
   const [isLoading, setIsLoading] = useState(true);
-
   const searchQueryParam = searchParams.get("search");
+  const filteredTag = searchParams.get("filterTag") as string;
+  const filteredCategory = searchParams.get("filterCategory") as string;
```
Comment (rebekahcallkacz) [inline src/components/SavedRecipes.tsx:29]: BLOCKING: For this and `filteredTag`, `.get()` can return a `string` or `null` so you need to handle both situations. You can't just cast directly to a string here. You'll need to handle the potential of a `null` value later on.

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get

File: src/components/SavedRecipes.tsx [hunk 2]
```diff
@@ -10,73 +20,123 @@ const SavedRecipes = () => {
...
   };
+  const categories = recipes.reduce(
+    (acc: Array<string>, recipe: SavedRecipe) => {
+      if (!acc.includes(recipe.recipeCategory)) {
+        acc.push(recipe.recipeCategory);
+      }
+      return acc;
+    },
+    []
+  );
```
Comment (rebekahcallkacz) [inline src/components/SavedRecipes.tsx:130]: This is a great use case for `reduce`!

**Review by rebekahcallkacz — CHANGES_REQUESTED** [review]
Only the one nitpicky comment - tested locally and works as expected!

Also, not sure who to note this to, but the tag for smoothies is misspelled (it should be smoothie, not smoothy).

**Comment by AnnaPestova1** [comment]
> Only the one nitpicky comment - tested locally and works as expected!
> 
> Also, not sure who to note this to, but the tag for smoothies is misspelled (it should be smoothie, not smoothy).

I will let our team know

File: src/components/SavedRecipes.tsx (hunk 1 above)
Comment (AnnaPestova1) [reply src/components/SavedRecipes.tsx:29]: Done!

**Review by rebekahcallkacz — APPROVED** [review]
LGTM

**Pull request merged**